hips/tiles/surveys.py

```diff
@@ -0,0 +1,67 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import urllib.request
+from astropy.table import Table
+from collections import OrderedDict
+from .description import HipsSurveyProperties
+
+__all__ = [
+    'HipsSurveyPropertiesList',
+]
+
+
+class HipsSurveyPropertiesList:
+    """HiPS survey container.
+
+    Parameters
+    ----------
+    surveys : list
+        HiPS surveys
+    """
+
+    def __init__(self, surveys: list) -> None:
+        self.surveys = surveys
+
+    @classmethod
+    def read(cls, filename: str) -> 'HipsSurveyPropertiesList':
+        """Read HiPS list from file (`HipsSurveyPropertiesList`).
+
+        Parameters
+        ----------
+        filename : str
+            HiPS list filename
+        """
+        with open(filename, 'r', encoding='utf-8', errors='ignore') as fh:
+            text = fh.read()
+        return cls.parse(text)
+
+    @classmethod
+    def fetch(cls, url: str='http://alasky.unistra.fr/MocServer/query?hips_service_url=*&dataproduct_type=!catalog&dataproduct_type=!cube&get=record') -> 'HipsSurveyPropertiesList':
+        """Fetch HiPS list text from remote location (`HipsSurveyPropertiesList`).
+
+        Parameters
+        ----------
+        url : str
+            HiPS list URL
+        """
+        text = urllib.request.urlopen(url).read()
+        return cls.parse(str(text))
+
+    @classmethod
+    def parse(cls, text: str) -> 'HipsSurveyPropertiesList':
+        """Parse HiPS list text (`HipsSurveyPropertiesList`).
+
+        Parameters
+        ----------
+        text : str
+            HiPS list text
+        """
+        surveys = list()
+        for raw_survey in text.split('\n\n'):
+            surveys.append(HipsSurveyProperties.parse(raw_survey).properties)
```

Comment:
I think the mypy error you mentioned might be due to this line.
You're calling `.properties` here, i.e. storing `OrderedDict` objects in the list.
What do you want to store as data member on `HipsSurveyPropertiesList`.
I would suggest you store `list` of `HipsSurveyProperties`, and not `list` of `OrderedDict`, i.e. don't remove the `.properties` here.